src/autotester/server/utils/tests/file_management_test.py

```diff
@@ -0,0 +1,264 @@
+        assert shutil.rmtree(dir, onerror=ignore_missing_dir_error)
+
+
+class TestMoveTree:
+    def test_empty_dir(self, empty_dir, dir_has_onedir):
+        """
+        When the Source Directory is empty
+        """
+        source_dir = empty_dir
+        dest_dir, sub_dir = dir_has_onedir
+        list_fd_bef_move = os.listdir(dest_dir)
+        move_tree(source_dir, dest_dir)
+        list_fd_after_move = os.listdir(dest_dir)
+        assert len(list_fd_bef_move) == len(list_fd_after_move)
+
+    def test_dir_has_onefile(self, empty_dir):
+        """
+        When the Source Directory has only one file
+        """
+        source_dir = tempfile.mkdtemp()
+        fd, source_file = tempfile.mkstemp(dir=source_dir)
+        dest_dir = empty_dir
+        move_tree(source_dir, dest_dir)
+        moved_file = os.path.join(dest_dir, os.path.basename(source_file))
+        assert os.path.exists(moved_file) and not os.path.exists(source_file)
+
+    def test_dir_has_onedir(self, empty_dir, dir_has_onedir):
+        """
+        When the Dirctory has only one sub directory
+        """
+        source_dir, sub_dir = dir_has_onedir
+        dest_dir = empty_dir
+        move_tree(source_dir, dest_dir)
+        moved_dir = os.path.join(dest_dir, os.path.basename(sub_dir))
+        assert os.path.exists(moved_dir) and not os.path.exists(sub_dir)
+
+    def test_dir_has_nestedfd(self, empty_dir, nested_fd):
+        """
+        When the files are nested in subdirectories more than 2 directories deep
+        """
+        dest_dir = empty_dir
+        source_dir = tempfile.mkdtemp()
+        sub_dir1 = tempfile.mkdtemp(dir=source_dir)
+        sub_dir2 = tempfile.mkdtemp(dir=sub_dir1)
+        fd, source_file = tempfile.mkstemp(dir=sub_dir2)
+        move_tree(source_dir, dest_dir)
+        moved_dir1 = os.path.join(dest_dir, os.path.basename(sub_dir1))
+        moved_dir2 = os.path.join(moved_dir1, os.path.basename(sub_dir2))
+        moved_file = os.path.join(moved_dir2, os.path.basename(source_file))
+        assert os.path.exists(moved_dir1) and not os.path.exists(sub_dir1)
+        assert os.path.exists(moved_dir2)
+        assert os.path.exists(moved_file)
+
+
+class TestFdOpen:
+    def test_open_dir(self):
+        with tempfile.TemporaryDirectory() as dir:
+            dir_fd = os.open(dir, os.O_RDONLY)
+            with fd_open(dir) as fdd:
+                same_dir = os.path.sameopenfile(fdd, dir_fd)
+            assert same_dir
+
+    def test_open_file(self):
+        with tempfile.NamedTemporaryFile() as file:
+            file_fd = os.open(file.name, os.O_RDONLY)
+            with fd_open(file.name) as fdf:
+                same_file = os.path.sameopenfile(fdf, file_fd)
+            assert same_file
+
+    def test_close(self):
+        with tempfile.TemporaryDirectory() as dir:
+            with fd_open(dir) as fdd:
+                dir_fd = fdd
+            with pytest.raises(Exception):
+                assert os.close(dir_fd)
```

Comment:
can you explain this check here please? I'm not sure this is doing what you think

Reply:
i am trying to close the dir which is already opened and closed by fd_open. Just checking if i try to close same file descriptor used by fd_open, it should raise an exception. correct me if i am checking this in a wrong way.

Reply:
I just think the assert is unnecessary in that case. Please review what the [`assert`](https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement) statement does and when it is [used when writing tests](https://docs.pytest.org/en/latest/assert.html)